libs/chain-signatures/contract/src/config/impls.rs

```diff
@@ -12,6 +13,35 @@ const MAX_EXPECTED_PARTICIPANTS: u32 = 32;
 /// that should be in the network.
 const NETWORK_MULTIPLIER: u32 = 128;
 
+// Default delay of five minutes, after that, request is removed from the contract state
+const DEFAULT_REQUEST_TIMEOUT_BLOCKS: u64 = 200;
+
+// The maximum number of requests to remove during a call
+const MAX_NUM_REQUESTS_TO_REMOVE: u32 = 100;
```

Comment:
Under the `GAS_FOR_SIGN_CALL` limit we need to account for the worst-case cost of removing the maximum number of requests. It would be helpful to have some estimate documented of how much that costs, especially if other parts of the contract change and `GAS_FOR_SIGN_CALL` needs to be readjusted in the future.

Reply:
Good point! Calling the `remove_timed_out_requests` endpoint and removing 100 signatures from the state requires approx. `37 Tgas`.

Calling `sign` and removing 100 signatures costs approximately `40 Tgas`. I left the the previous gas requirement of 50 TGAS in place to have additional margin and added tests that ensure a signature call does not exceed that limit.

Reply:
@kevindeforth no we should not require 50Tgas for each sign call because it means that when `sign` is called from other smart contract, they need to attach at least 50Tgas to the promise and that is too high and limits what the caller contract can do. We should make it as low as possible. One way to do it is to remove 1 request from the state state when `sign` is called.

Reply:
Understood. The following is the behavior as of the latest commit:
- Gas requirement for sign call is 10 Tgas.
- Default for `config.max_num_requests_to_remove` is set to 1.
- In the pytests, we now attach only 10 TGas to the sign transactions (as opposed to the previous 150), to reflect how users are expected to interact with it.
- The `remove_timed_out_requests` contract endpoint takes an additional argument `max_num_to_remove: Option<u32>`. The function serves as a back-up in case we need to manually clear the state.